flake.nix

```diff
@@ -0,0 +1,109 @@
+{
+  description = "A Nix flake for OSRD dev shell";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
```

Comment:
rust-overlay also has a nixpkgs input. It might be best to either make nixpkgs follow rust-overlay's, or the other way around. As is, this downloads two versions of nixpkgs